mgo.go

```diff
@@ -248,7 +248,7 @@ func (inst *MgoInstance) run() error {
 		return err
 	}
 	logger.Debugf("found mongod at: %q", mongopath)
-	if mongopath == "/usr/lib/juju/bin/mongod" {
+	if mongopath == "/usr/lib/juju/bin/mongod" || mongopath == "/usr/lib/juju/mongo3.2/bin/mongod" {
```

Comment:
Given the differences in 2.4 vs 2.6 we weren't able to support them generically even with simple version changes. So we decided to explicitly version the mongo directory, so that new versions of Juju don't accidentally get a version of Mongo that they can't actually support.

Since we have to do work to enable Mongo anyway, we'd have to release new versions to get new Mongo. (btw, Mongo uses even/odd for stable/unstable, so we'll never do 3.3, but certainly we should be aware of what 3.4 might bring.)